resources/js/components/Checkout/Checkout.vue

```diff
@@ -27,13 +27,20 @@ export default {
         this.checkout.hasVirtualItems = this.hasVirtualItems
         this.steps = this.config.checkout_steps[window.config.store_code] ?? this.config.checkout_steps['default']
         this.setupHistory()
-        this.setCheckoutCredentialsFromDefaultUserAddresses()
-        this.getShippingMethods()
-        this.getTotalsInformation()
-        this.$root.$emit('checkout-step', 1)
+        this.setupCheckout()
     },
 
     methods: {
+        async setupCheckout() {
+            await this.refreshUser()
+
+            this.setCheckoutCredentialsFromDefaultUserAddresses()
+            this.getShippingMethods()
+            this.getTotalsInformation()
+
+            this.$root.$emit('checkout-step', 1)
```

Comment:
Do we need all these functions to wait until refreshUser is done, or are there some functions which can run independently of the refreshUser function?

If so it may be better to use .then and add the functions dependent on user info within the .then.
And remove the await.

This way no functions are obstructed from running until the user info call has finished

Reply:
Unfortunately we need the default credentials function to run before getting the shipping information and totals information, and for the default credentials we need this user refresh. We also don't want to emit the `checkout-step` event before these addresses are set correctly, so this is about as good as it gets here.